tests/tests_pytorch/trainer/connectors/test_accelerator_connector.py

```diff
@@ -369,6 +369,33 @@ def test_exception_invalid_strategy():
         Trainer(strategy="tpu_spawn")
 
 
+@pytest.mark.parametrize(
+    ["strategy", "strategy_class"],
+    (
+        ("ddp_spawn", DDPSpawnStrategy),
+        ("ddp_spawn_find_unused_parameters_false", DDPSpawnStrategy),
+        ("ddp", DDPStrategy),
+        ("ddp_find_unused_parameters_false", DDPStrategy),
+        ("dp", DataParallelStrategy),
+        ("ddp_sharded", DDPShardedStrategy),
+        ("ddp_sharded_spawn", DDPSpawnShardedStrategy),
+    ),
+)
+@RunIf(mps=True)
+def test_exception_invalid_ddp_strategy_with_mps(strategy, strategy_class):
+    # check with str
+    with pytest.raises(
+        MisconfigurationException, match=r"With `accelerator=mps`, strategies from DDP Family are not compatible"
+    ):
+        Trainer(accelerator="mps", strategy=strategy)
```

Comment:
Edge case alert! We also need to handle `accelerator="auto"` :))

Reply:
and `accelerator="gpu"` but at some point they will be set to "mps" , iirc

Reply:
I have added a case for the `accelerator="auto"` condition. For the edge case @justusschock mentioned, there are other test cases that fail when I add another check. I have commented it out for now,
```
        # MPSAccelerator incompatible with ddp-family
        # check if strategy is string and contains 'ddp' in it
        is_ddp_str = isinstance(strategy, str) and "ddp" in strategy
        # MPS accelerator could be instantiated in 3 ways,
        # 1. accelerator = "mps"
        # 2. accelerator in ("auto", None) and MPSAccelerator.is_available()
        # 3. accelerator = "gpu" and MPSAccelerator.is_available()
        is_mps_accelerator = (
            (isinstance(accelerator, str) and accelerator == "mps")
            or (
                ((isinstance(accelerator, str) and accelerator == "auto") or accelerator is None)
                and MPSAccelerator.is_available()
            )
            # or (isinstance(accelerator, str) and accelerator == "gpu" and MPSAccelerator.is_available())
        )
        if is_mps_accelerator and (is_ddp_str or isinstance(strategy, ParallelStrategy)):
            raise ValueError("With MPSAccelerator, strategies from the DDP family are not compatible.")
```
Can you give some feedback as to why you think the commented-out section causes some test cases to fail?

Reply:
@shenoynikhil Finally I got to look at this again. I handled the additional edge case and new ones too, and simplified the boolean logic as much as possible. I updated tests that needed updating. 

Notably: Some tests that only instantiated Trainer but not run fit would now error with our newly added validation, when run on an M1 mac. I handled all these cases by either simulating no mps devices in the test, or skipping the test on mps hardware completely.

Reply:
> Some tests that only instantiated Trainer but not run fit would now error with our newly added validation, when run on an M1 mac.

That makes sense. I noticed these when I was writing tests. Wasn't sure if I should modify the others. Skipping them on mps hardware makes sense.